evm/vm/base.py

```diff
@@ -254,20 +258,21 @@ def configure_header(self, **header_params):
     def snapshot(self):
         """
         Perform a full snapshot of the current state of the VM.
-
-        TODO: This needs to do more than just snapshot the state_db but this is a start.
         """
-        with self.state_db(read_only=True) as state_db:
-            return state_db.snapshot()
+        return (self.block.header.state_root, self.journal_db.snapshot())
 
     def revert(self, snapshot):
         """
         Revert the VM to the state
-
-        TODO: This needs to do more than just snapshot the state_db but this is a start.
         """
+        state_root, journal_index = snapshot
+
         with self.state_db() as state_db:
-            return state_db.revert(snapshot)
+            state_db.root_hash = state_root
+            self.journal_db.revert(journal_index)
+
+    def commit(self):
```

Comment:
This API feels awkward.  I'm not sure what to rename it to or how to fix it right now.